pyomo/contrib/pynumero/interfaces/poeknl_nlp.py

```diff
@@ -0,0 +1,242 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+"""
+This module defines the classes that provide an NLP interface based on
+the Ampl Solver Library (ASL) implementation
+"""
+from pyomo.contrib.pynumero.interfaces.ampl_nlp import AslNLP
+import pyutilib
+
+from scipy.sparse import coo_matrix
+import numpy as np
+import six
+
+try:
+    import poek as pk
+    poek_available=True
+    __all__ = ['PoekNLP_NLP']
+except:
+    poek_available=False
+    __all__ = []
+
+
+class PoekNL_NLP(AslNLP):
```

Comment:
My guess is that PoekNL_NLP is mostly for testing performance. Is that correct? It would be very interesting to see how does performance of the AD from Poek compare against ASL. If the AD of Poek is as fast or close as fast to ASL then PoekNLP would be always a better choice than PoekNL_NLP? It would also be interesting to see how does the nl writer of Poek compares to the one in Pyomo (you probably have already some metrics on that?)

Its cool seeing that pynumero is facilitating this comparisons :)